### ISSUE TITLE
Write a JSON API

### ISSUE THREAD

**Issue by J3RN** [J3RN/timesheet#44]
It would be excellent to have a JSON API so I can create a mobile app and CLI tool.

**Comment by J3RN** [comment]
Additionally, I intend to create a "fat client" JavaScript version because people think that's real hot and it will look good on my résumé.

For reference, I intend to upgrade to Rails 5.1 before starting this, but don't really want to create more clutter by opening another issue for it.
